keepsorted/keep_sorted_test.go

```diff
@@ -39,6 +41,14 @@ var (
 	}
 )
 
+// initZerolog initializes zerolog to log as part of the test.
+// It returns a function that restores zerolog to its state before this function was called.
+func initZerolog(t testing.TB) func() {
+	oldLogger := log.Logger
+	log.Logger = log.Output(zerolog.NewTestWriter(t))
+	return func() { log.Logger = oldLogger }
```

Comment:
Nit: use `t.Cleanup(func() { log.Logger = oldLogger })` instead of returning a function

Reply:
oh neat, `t.Cleanup` even does LIFO so if someone called `initZerolog` multiple times it'd still kinda not break everything.